### PR TITLE
save data in JLD2 format

### PR DESCRIPTION
**Pull request by Eben60**
**Save data in JLD2 format, add isotope symbols (e.g. ³He), combine values and uncertainties into Measurements (e.g. 6.04496±0.00027 u ), export to CSV.**

Julia Serialization is not recommended for data storage, as the data are stored in an internal format and may change between OSes and Julia versions. JLD2 format allows to keep the data structure just like Serialization does, but adds compatibility. 

The first script just reads the serialized data and re-saves them as JLD2. The second and third add symbols and combine columns into Measurement typed data. The fourth exports data to CSV - @lmmentel , you may want to use these for your Mendeleev package.

The JLD2 and CSV files thus produced are in the same directory.

@Gregstrq , what do you think of publishing a bare-bones Julia package consisting of the files `IsotopesData.jl` and `isotopes_data_measvals.jld` ? Actually I believe the not too numerous people needing the data ( @TimSlendebroek  , anybody else?) would be happy enough with the data in DataFrames format. Publishing a package would simplify the use and add visibility. Later on it is always possible to create a dependent package with all syntactic sugar one can think of. Or should I take over creating the package?

### CODE REVIEW

**Comment by Gregstrq** [comment]
I see no problems with saving the main dataframe into jld2 instead of jls file. I also like idea with combining value and uncertainty into the single measurement (I was not really aware of [Measurements.jl](https://github.com/JuliaPhysics/Measurements.jl)).
In that regards, I would still suggest a few tweaks:
- We need to be more careful with the way we treat the situation where the value is non-`missing`, while the uncertainty is `missing`. I've checked, and currently all such occurences correspond to the data on half lives. Physically, there are two distinct situations here: there are stable isotopes which have infinite lifetime, and there are the estimates for the half-life without the normal uncertainty. I suggest we make the uncertainty zero in the former case, and NaN in the latter case:
```
measmiss(v, err::Missing) = isinf(v) ? measurement(v, 0.0*unit(v)) : measurement(v, NaN*unit(v))
```
It may not be the best solution, but it makes easier to filter out these cases and alter them later.

- When you are saving the CSV file, I suggest you alter the names of the unitful fields to contain the unit. For example, the name of `quadrupole_moment` field can be "quadrupole moment [b]". See the header of "elec_mom_recomm.csv" as an example.
- I don't like proliferation of files. I would simply alter the end of "parse_isotopes.jl", so that when it is run, it would generate both .csv and .jld2 files. You can save to .csv, transform dataframe to add measurements and save to .jld2 in place of saving to .jls at the end of that file.
- IsotopesData.jl does not belong here. This repo is only related to generating the isotopic data. But I hope to publish "Isotopes.jl" soon, and we can discuss the structure of the package providing the data there.

**Comment by Gregstrq** [comment]
The isotopic symbols ³He is a separate matter. It is very nice for pretty printing, but internally it does not matter (it does not add any new information about an isotope), and I am not sure it is a pragmatic choice if one wants to select an isotope.

Let's say the user wants to run something like `isotopes(symbol)` and get the isotopic data. Writing `isotopes(:He3)` is faster then doing `isotopes("\^3\tabHe")`. (You have to type in `\^3\tab` to create superscript, and you need to have it as a string.)

We can keep this field, but, as I said, it will mostly be used for pretty printing.

**Comment by Eben60** [comment]
> I suggest we make the uncertainty zero in the former case, and NaN in the latter case.

Yes, thank you, agree, I think it's the best we can do given the limitations of `Measurements`. I've updated accordingly.

> The isotopic symbols ³He is a separate matter. It is very nice for pretty printing, but internally it does not matter...
> We can keep this field, but, as I said, it will mostly be used for pretty printing.

Sure, it was meant for pretty printing only - but that's reason enough to have it.

> When you are saving the CSV file, I suggest you alter the names of the unitful fields to contain the unit. 

I do not thing there are going to be a lot of users of this CSV file - maybe @lmmentel , but who else? Thus I do not think it's worth changing the headers, it's easier to tell explicitly, the masses in dalton, half-times in second, and quadrupol moments in barn. I however added a small text file `isotopes_data_units.txt` with exactly this information.

> I don't like proliferation of files. 

What we have here is more or less an intermediate state, and I personally do not thing it's worthwhile perfectioning the workflows which are not going to be used any often (well, which are actually already done). If you don't like having intermediate files - delete them and forget them. 

> IsotopesData.jl does not belong here. This repo is only related to generating the isotopic data. But I hope to publish "Isotopes.jl" soon...

It was the suggestion to publish the data in this form as separate package. Are you explicitly against it? And, are you explicitly against me publishing the data in this form (with all due credits to your work, of course)?

**Comment by lmmentel** [comment]
@Eben60 thanks for the csv file :raised_hands:  I however was missing some of the details on how the values were actually put together so I started looking into the data myself and got sucked in to a point where I now updated the data on isotopes in [mendeleev](https://github.com/lmmentel/mendeleev) in [this PR](https://github.com/lmmentel/mendeleev/pull/74). I'm still validating but I'm sharing here as an FYI. The updates should go into the next release which I'm planning soon.

@Gregstrq thanks for putting this repo togehter, unfortunately I wasn't able to read the data in, and got this error

```julia
julia> isotopes = deserialize("isotopes_data.jls")
ERROR: ArgumentError: array must be non-empty
Stacktrace:
 [1] pop! at ./array.jl:1126 [inlined]
 ```

My julia chops and not great, but maybe you could give me a hint here?

**Comment by Eben60** [comment]
> My julia chops and not great, but maybe you could give me a hint here?

The documentation of `Serialization` states that there could be problems if the configurations of reading and writing machine differ (Juli version, OS, packages etc.). That was exactly the reason for me to convert the data into JLD2 format.

**Comment by Gregstrq** [comment]
Sorry for not responding for a long time. I was on a vacation.
> It was the suggestion to publish the data in this form as separate package. Are you explicitly against it?

Publishing it as a separate package is indeed my plan. 

> And, are you explicitly against me publishing the data in this form (with all due credits to your work, of course)?

Let's be honest. I would prefer to publish it myself. If I ever decide to switch to industry, an additional package on my github page would be very beneficial.

File: JLD2_saved/03_isotopes2measvals.jl [hunk 1]
```diff
@@ -11,11 +11,20 @@ function getmeasnames(df)
...
 measmiss(v::Missing, err) = missing
 measmiss(v::Missing, err::Missing) = missing
 # as some half-life data miss an estimation of uncertainty, and measurement(::Float64, Missing) is not defined
-measmiss(v, err::Missing) = v
+# measmiss(v, err::Missing) = isinf(v) ? (v ± zero(v)) : (v ± convert(typeof(v), NaN))
+
+function measmiss(v, err::Missing)
+    isinf(v) && return (v ± zero(v))
+
+    typednan = convert(typeof(ustrip(v)), NaN)
```
Comment (Gregstrq) [inline JLD2_saved/03_isotopes2measvals.jl:23]: This is actually an overkill. You can not convert `NaN` to `Int`, only to `Float32` at best.

**Review by Gregstrq — APPROVED** [review]

**Pull request merged**